### PR TITLE
Stop using deprecated templating service

### PR DESCRIPTION
**Pull request by MKodde**
A PR in a similar vain to #98 only here we simply update the reference to the twig templating engine to the current standard. Here is my rationale supporting this change:

1. Template logic should be separated from controller logic for improved maintainability and predictability.
2. All OpenConext products using this bundle utilize the Twig template engine.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -7,22 +7,24 @@
...
         "php": ">=7.2,<8.0-dev",
+        "ext-dom": "*",
         "ext-openssl": "*",
         "robrichards/xmlseclibs": "^3.0.4",
         "simplesamlphp/saml2": "3.2.*",
         "symfony/dependency-injection": "^4.4",
         "symfony/framework-bundle": "^4.4",
         "symfony/templating": "^4.4",
-        "ext-dom": "*"
+        "twig/twig": "^2"
```
Comment (MKodde) [inline composer.json:17]: In order to run the tests, we need Twig to be able to render the metadata template

File: src/Tests/Component/Metadata/MetadataFactoryTest.php [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+XML;
+        $metadataConfiguration = new MetadataConfiguration();
+        $metadataConfiguration->isIdP = true;
+        $metadataConfiguration->idpCertificate = __DIR__ . '/keys/idp-cert.pem';
+        $metadataConfiguration->ssoRoute = 'https://foobar.example.com';
+        $metadataConfiguration->entityIdRoute = 'https://foobar.example.com';
+        $this->buildFactory($metadataConfiguration);
+        $metadata = $this->factory->generate();
+        $constraint = new  XSDValidation(__DIR__ . '/xsd/metadata.xsd');
```
Comment (MKodde) [inline src/Tests/Component/Metadata/MetadataFactoryTest.php:117]: Running the XSD schema validation is not the fastest test on the planet, so it is used sparingly. 

A download of the XSD schema and it's derivatives was required to run the test. As the other schemas would be curled from the internet into the test, which timed out the test as some of these schemas where 404-ing

Comment (thijskh) [reply src/Tests/Component/Metadata/MetadataFactoryTest.php:117]: Can we just ship the schemas in git? They will never really change?

**Review by thijskh — APPROVED** [review]

File: src/Tests/Component/Metadata/MetadataFactoryTest.php (hunk 2 above)
Comment (MKodde) [reply src/Tests/Component/Metadata/MetadataFactoryTest.php:117]: Yup that solved the timeout/404 problems. I agree that keeping them local is safe enough. And updating them is simple enough.

**Pull request merged**